cli/compose/interpolation/interpolation.go

```diff
@@ -67,7 +67,10 @@ func recursiveInterpolate(value any, path Path, opts Options) (any, error) {
 			return newValue, nil
 		}
 		casted, err := caster(newValue)
-		return casted, newPathError(path, errors.Wrap(err, "failed to cast to expected type"))
+		if err != nil {
+			return casted, newPathError(path, errors.Wrap(err, "failed to cast to expected type"))
```

Comment:
Wasn't sure if `caster` would still return a value, so I left that one as-is, and just return it (as it did before); it should probably be `nil` or "something else" though.